pynecone/components/graphing/victory.py

```diff
@@ -0,0 +1,567 @@
+    else:
+        return [{"x": x[i], "y": y[i], "amount": amount[i]} for i in range(len(x))]
+
+
+def format_area(x: List, y: List, y0: Optional[List] = None) -> List:
+    """Format area data.
+
+    Args:
+        x: The x values.
+        y: The y values.
+        y0: The y0 values.
+
+    Returns:
+        The formatted data.
+
+    Raises:
+        ValueError: If x and y are not the same length.
+    """
+    if y0 is None:
+        return format_xy(x, y)
+    else:
+        if len(x) != len(y) or len(x) != len(y0):
+            raise ValueError("x, y, and y0 must be the same length")
+        return [{"x": x[i], "y": y[i], "y0": y0[i]} for i in range(len(x))]
+
+
+def format_bar(x: List, y: List, y0: Optional[List] = None) -> List:
+    """Format bar data.
+
+    Args:
+        x: The x values.
+        y: The y values.
+        y0: The y0 values.
+
+    Returns:
+        The formatted data.
+
+    Raises:
+        ValueError: If x and y are not the same length.
+    """
+    if y0 is None:
+        return format_xy(x, y)
+    else:
+        if len(x) != len(y) or len(x) != len(y0):
+            raise ValueError("x, y, and y0 must be the same length")
+        return [{"x": x[i], "y": y[i], "y0": y0[i]} for i in range(len(x))]
+
+
+def format_box_plot(
+    x: List,
+    y: Optional[List[Any]] = None,
+    min_: Optional[List[Any]] = None,
+    max_: Optional[List[Any]] = None,
+    median: Optional[List[Any]] = None,
+    q1: Optional[List[Any]] = None,
+    q3: Optional[List[Any]] = None,
+) -> List:
+    """Format box plot data.
+
+    Args:
+        x: The x values.
+        y: The y values.
+        min_: The minimum values.
+        max_: The maximum values.
+        median: The median values.
+        q1: The q1 values.
+        q3: The q3 values.
+
+    Returns:
+        The formatted data.
+
+    Raises:
+        ValueError: If x is not provided.
+        ValueError: If y is provided and x, y are not the same length.
+        ValueError: If y is not provided and min, max, median, q1, and q3 are not provided.
+        ValueError: If y is not provided and x, min, max, median, q1, and q3 are not the same length.
+    """
+    data = []
+    if x is None:
+        raise ValueError("x must be specified")
+
+    if y is not None:
+        return format_xy(x, y)
+
+    else:
+        if min_ is None or max_ is None or median is None or q1 is None or q3 is None:
+            raise ValueError(
+                "min, max, median, q1, and q3 must be specified if y is not provided"
+            )
+        elif (
+            len(x) != len(min_)
+            or len(x) != len(max_)
+            or len(x) != len(median)
+            or len(x) != len(q1)
+            or len(x) != len(q3)
+        ):
+            raise ValueError(
+                "x, min, max, median, q1, and q3 must be the same length and specified if y is not provided"
+            )
+        for i in range(len(x)):
+            row = {}
+            if x is not None:
```

Comment:
Don't you already check that these are not `None` above?

Reply:
Ah yeah that was from before when I was making it so you could pass in some of them but changed it to need all if y inset specified